dali/operators/random/rng_base_gpu.h

```diff
@@ -27,23 +27,19 @@
 
 namespace dali {
 
-template <bool IsNoiseGen>
-struct BlockDesc;
-
-template <>
-struct BlockDesc<false> {
-  int sample_idx;
-  void* output;
-  size_t size;
+struct SampleDesc {
+  void *output;
+  const void* input;
+  int64_t p_count;
+  int64_t p_stride;
+  int64_t c_count;
+  int64_t c_stride;
 };
 
-template <>
-struct BlockDesc<true> {
+struct BlockDesc {
   int sample_idx;
-  void* output;
-  const void* input;
-  size_t offset;
-  size_t size;
+  int64_t p_offset;
+  int64_t p_count;
```

Comment:
Does BlockDesc.p_count mean the same as SampleDesc.p_count ?

Reply:
The same concept, but one is the p_count (pixel count, point count, I didn't call it pixel because this is not limited to images) for the whole sample, and the other one is a part of it (that's what a block is, an offset and a number of points). Depending if it's per-channel or not we are flattening the channels together with the_p_count, or treat whole pixels as 1.